velox/exec/fuzzer/AggregationFuzzerBase.cpp

```diff
@@ -318,8 +326,21 @@ std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputDataWithRowNumber(
           generator->generate(signature.args, vectorFuzzer_, rng_, pool_.get());
     }
 
+    // Number of partitions is randomly generated and is at least 1.
+    auto numPartitions = size ? randInt(1, size) : 1;
+    auto indices = vectorFuzzer_.fuzzIndices(size, numPartitions);
     for (auto i = children.size(); i < types.size() - 1; ++i) {
-      children.push_back(vectorFuzzer_.fuzz(types[i], size));
+      if (partitionKeySet.find(names[i]) != partitionKeySet.end()) {
+        // The partition keys are built with a dictionary over a smaller set of
+        // values. This is done to introduce some repetition of key values for
+        // windowing.
+        auto baseVector = vectorFuzzer_.fuzz(types[i], numPartitions);
+        auto nulls = vectorFuzzer_.fuzzNulls(size);
```

Comment:
We should align nulls as well for all the partition columns I feel. So it might be better to move this along with the indices at line 331.